src/category_theory/bicategory/basic.lean

```diff
@@ -0,0 +1,468 @@
+  (f ◁ θ) ≫ (η ▷ i) = (η ▷ h) ≫ (g ◁ θ) . obviously)
+-- associator:
+(associator {a b c d : B} (f : a ⟶ b) (g : b ⟶ c) (h : c ⟶ d) :
+  (f ≫ g) ≫ h ≅ f ≫ (g ≫ h))
+(notation `α_` := associator)
+(associator_naturality_left' :
+  ∀ {a b c d} {f f' : a ⟶ b} (η : f ⟶ f') (g : b ⟶ c) (h : c ⟶ d),
+  ((η ▷ g) ▷ h) ≫ (α_ f' g h).hom = (α_ f g h).hom ≫ (η ▷ (g ≫ h)) . obviously)
+(associator_naturality_middle' :
+  ∀ {a b c d} (f : a ⟶ b) {g g' : b ⟶ c} (η : g ⟶ g') (h : c ⟶ d),
+  ((f ◁ η) ▷ h) ≫ (α_ f g' h).hom = (α_ f g h).hom ≫ (f ◁ (η ▷ h)) . obviously)
+(associator_naturality_right' :
+  ∀ {a b c d} (f : a ⟶ b) (g : b ⟶ c) {h h' : c ⟶ d} (η : h ⟶ h'),
+  ((f ≫ g) ◁ η) ≫ (α_ f g h').hom = (α_ f g h).hom ≫ (f ◁ (g ◁ η)) . obviously)
+--left unitor:
+(left_unitor {a b : B} (f : a ⟶ b) : 𝟙 a ≫ f ≅ f)
+(notation `λ_` := left_unitor)
+(left_unitor_naturality' : ∀ {a b} {f f' : a ⟶ b} (η : f ⟶ f'),
+  (𝟙 a ◁ η) ≫ (λ_ f').hom = (λ_ f ).hom ≫ η . obviously)
+-- right unitor:
+(right_unitor {a b : B} (f : a ⟶ b) : f ≫ 𝟙 b ≅ f)
+(notation `ρ_` := right_unitor)
+(right_unitor_naturality' : ∀ {a b} {f f' : a ⟶ b} (η : f ⟶ f'),
+  (η ▷ 𝟙 b) ≫ (ρ_ f').hom = (ρ_ f ).hom ≫ η . obviously)
+-- pentagon identity:
+(pentagon' : ∀ {a b c d e} (f : a ⟶ b) (g : b ⟶ c) (h : c ⟶ d) (i : d ⟶ e),
+  ((α_ f g h).hom ▷ i) ≫ (α_ f (g ≫ h) i).hom ≫ (f ◁ (α_ g h i).hom)
+  = (α_ (f ≫ g) h i).hom ≫ (α_ f g (h ≫ i)).hom . obviously)
+-- triangle identity:
+(triangle' : ∀ {a b c} (f : a ⟶ b) (g : b ⟶ c),
+  (α_ f (𝟙 b) g).hom ≫ (f ◁ (λ_ g).hom) = (ρ_ f).hom ▷ g . obviously)
+
+restate_axiom bicategory.whisker_left_id'
+attribute [simp] bicategory.whisker_left_id
+restate_axiom bicategory.whisker_left_comp'
+attribute [reassoc, simp] bicategory.whisker_left_comp
+restate_axiom bicategory.whisker_right_id'
+attribute [simp] bicategory.whisker_right_id
+restate_axiom bicategory.whisker_right_comp'
+attribute [reassoc, simp] bicategory.whisker_right_comp
+restate_axiom bicategory.whisker_exchange'
+attribute [simp, reassoc] bicategory.whisker_exchange
+restate_axiom bicategory.associator_naturality_left'
+attribute [reassoc] bicategory.associator_naturality_left
+restate_axiom bicategory.associator_naturality_middle'
+attribute [reassoc] bicategory.associator_naturality_middle
+restate_axiom bicategory.associator_naturality_right'
+attribute [reassoc] bicategory.associator_naturality_right
+restate_axiom bicategory.left_unitor_naturality'
+attribute [reassoc] bicategory.left_unitor_naturality
+restate_axiom bicategory.right_unitor_naturality'
+attribute [reassoc] bicategory.right_unitor_naturality
+restate_axiom bicategory.pentagon'
+attribute [reassoc] bicategory.pentagon
+restate_axiom bicategory.triangle'
+attribute [simp, reassoc] bicategory.triangle
+
+infixr ` ◁ `:70 := bicategory.whisker_left
+infixr ` ▷ `:70 := bicategory.whisker_right
+notation `α_` := bicategory.associator
+notation `λ_` := bicategory.left_unitor
+notation `ρ_` := bicategory.right_unitor
```

Comment:
Should we make this notation localized? I assume that there might be other places where we want to use this notation that doesn't fit exactly into this pattern of bicategories.